Branch-SDK/Branch-SDK/BranchViewHandler.m

```diff
@@ -105,6 +107,28 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
     return !isRedirectHandled;
 }
 
+- (void)webViewDidFinishLoad:(UIWebView *)webView {
+    NSCachedURLResponse *resp = [[NSURLCache sharedURLCache] cachedResponseForRequest:webView.request];
+    NSInteger status = [(NSHTTPURLResponse*)resp.response statusCode];
+    NSLog(@"Status: %ld",(long)status);
+    if (status == 200) {
+        if (self.pendingBranchView != nil && self.pendingWebview != nil) {
+            UIViewController *holderView = [[UIViewController alloc] init];
+            [holderView.view insertSubview:self.pendingWebview atIndex:0];
+            UIViewController *presentingViewController = [[[[UIApplication sharedApplication] windows] firstObject] rootViewController];
+            [presentingViewController presentViewController:holderView animated:YES completion:nil];
+            
+            [self.pendingBranchView updateUsageCount];
+            
+            if (self.branchViewCallback) {
+                [self.branchViewCallback branchViewVisible:self.pendingBranchView.branchViewAction withID:self.pendingBranchView.branchViewID];
+            }
+        }
+    }
```

Comment:
Hi Derrick,

The objective, as explained to me, is to avoid having a blank white view
appear on the screen should the request result in any error.
Now, only if the page load is successful, will it be displayed.
Maybe Sojan can provide more detail.  I don't yet have enough knowledge
about the system to suggest what we may want to do to address errors.

Phillip



On Tue, Apr 19, 2016 at 5:15 PM, derrickstaten <notifications@github.com>
wrote:

> In Branch-SDK/Branch-SDK/BranchViewHandler.m
> <https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/pull/337#discussion_r60333883>
> :
>
> > +    NSInteger status = [(NSHTTPURLResponse*)resp.response statusCode];
> > +    NSLog(@"Status: %ld",(long)status);
> > +    if (status == 200) {
> > +        if (self.pendingBranchView != nil && self.pendingWebview != nil) {
> > +            UIViewController *holderView = [[UIViewController alloc] init];
> > +            [holderView.view insertSubview:self.pendingWebview atIndex:0];
> > +            UIViewController *presentingViewController = [[[[UIApplication sharedApplication] windows] firstObject] rootViewController];
> > +            [presentingViewController presentViewController:holderView animated:YES completion:nil];
> > +
> > +            [self.pendingBranchView updateUsageCount];
> > +
> > +            if (self.branchViewCallback) {
> > +                [self.branchViewCallback branchViewVisible:self.pendingBranchView.branchViewAction withID:self.pendingBranchView.branchViewID];
> > +            }
> > +        }
> > +    }
>
> it's not entirely clear to me how this works end to end, but the lack of
> an error handler worries me a bit. is there a case where the developer
> would want to know if status != 200?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/pull/337/files/3480fc78dbfe993898ffed68051116b6357e4808#r60333883>
>